### ISSUE TITLE
[remote] command line option for new `trackerList` RPC API

### ISSUE THREAD

**Issue by tearfur** [transmission/transmission#7135]
### What is the issue?

Sparked from #7134.

`trackerAdd` and `trackerRemove` are deprecated API but still used in `transmission-remote`, we need new command line option(s) that uses the new `trackerList` API to replace them.

### Which application of Transmission?

transmission-remote

### Which version of Transmission?

ed2c6c40851b2c8cdd20b3a99c1f823898bdd055

**Comment by killemov** [comment]
Using `trackerList` for adding/removing specific trackers works fine for single torrent use only. For multiple torrents it is very inefficient because you need to iterate over all torrents (client side!) and get/update/set the `trackerList` for each of them.

The original fields `trackerAdd`, `trackerRemove` and `trackerReplace` in `torrent-set` did have some advantage for batch updates but were implemented very badly as in quit when a duplicate was being added or a non-existent tracker was being removed. Adding insult to injury `trackerRemove` and `trackerReplace` were implemented using indexes instead of announce strings. Sigh.

I suggest implementing some new tracker-centric top level methods that are tailored for batch updates.

- `tracker-get` to get tracker fields for specific torrent ids.
- `tracker-set` to set tracker fields for specific torrent ids.
- `tracker-search` to find all torrents that have some specific trackers.
- `tracker-replace` to replace some specific trackers.

None of these methods should use any `id` to identify a tracker. All of these methods could use `torrentIds` to limit the scope of the operations.

See also #3226, #6267 and #6964. (Non-exhaustive list of references.)

**Comment by killemov** [comment]
> Can't they co-exist?

No. The original fields are actually methods/actions and should be implemented as such. They were also likely meant for single torrent use even though this was not explicitly documented. The halting on adding duplicates, halting on removing non-existents and the use on indexes leads me to believe this. So there never was a proper way to do batch updates anyway. I just implemented it in Shift thinking there was. The RPC-spec was not clear about it.

> so I'm curious about the reason to deprecate them.

They were correctly superseded by `trackerList` and should never return as fields for `set-torrent`. (Even though I first objected to this change and @ckerr kind of agreed with me.) Also the `trackerX` fields were implemented badly from the start anyway. However implementing the methods I mentioned before would massively benefit batch updating trackers. (Change to snake_case if you must.)

**Comment by tearfur** [comment]
> The original fields are actually methods/actions and should be implemented as such.

The list of new methods can be further refined, but I can agree with this in general. Personally I'd prefer namings like `torrent-add-tracker`, similar to `torrent-set-location`.

Regardless, for anyone wishing to work on this issue, the scope will still be limited to adding a new command line parameter for `trackerLists`. Please treat this discussion as extra.

> They were correctly superseded by `trackerList` and should never return as fields for `set-torrent`.

Right, after reading your explanation it makes sense to me.

**Comment by Flashlamenace** [comment]
Is this issue taken ? can I do it ?
